### PR TITLE
net/bluetooth: Bluetooth supported commands

### PR DESCRIPTION
**Pull request by amitpurwar**

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [b0423093d23dc4f3ec4c20413cbc60b4e1ae34b1] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [b0423093d23dc4f3ec4c20413cbc60b4e1ae34b1] - Code Rule Check (C++) OK.

File: os/net/bluetooth/bt_util.h [hunk 1]
```diff
@@ -131,4 +131,154 @@
...
+ *  Get a 16-bit integer, stored in big-endian format in a potentially
+ *  unaligned memory location, and convert it to the host endianness.
+ *
+ *  @param src Location of the big-endian 16-bit integer to get.
+ *
+ *  @return 16-bit integer in host endianness.
+ */
+static inline uint16_t sys_get_be16(const uint8_t src[2])
+{
+	return ((uint16_t) src[0] << 8) | src[1];
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_util.h]: Please remove space in typecasting.

like: ((uint16_t)src[0] << 8) | src[1];

File: os/net/bluetooth/bt_util.h [hunk 2]
```diff
@@ -131,4 +131,154 @@
...
+ *  Get a 32-bit integer, stored in big-endian format in a potentially
+ *  unaligned memory location, and convert it to the host endianness.
+ *
+ *  @param src Location of the big-endian 32-bit integer to get.
+ *
+ *  @return 32-bit integer in host endianness.
+ */
+static inline uint32_t sys_get_be32(const uint8_t src[4])
+{
+	return ((uint32_t) sys_get_be16(&src[0]) << 16) | sys_get_be16(&src[2]);
```
Comment (NilabhShant) [inline os/net/bluetooth/bt_util.h]: same as above and please check this at all instances.

File: os/net/bluetooth/bluetooth.c [hunk 3]
```diff
@@ -301,6 +301,63 @@ static int bt_conn_init(void)
...
 
+static void le_read_supp_states_complete(struct bt_buf_s *buf)
+{
+        struct bt_hci_rp_le_read_supp_states_s *rp = (void *)buf->data;
+
+        ndbg("status %u", rp->status);
+
+        g_btdev.le.states = sys_get_le64(rp->le_states);
+}
+
```
Comment (drashti304) [inline os/net/bluetooth/bluetooth.c]: Please remove extra blank lines

Comment (amitpurwar) [reply os/net/bluetooth/bluetooth.c]: removed the extra line

**Comment by seinfra** [comment]
Target : [32d746ccff2925ad42d12c62cae3c2963bc8d271] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [32d746ccff2925ad42d12c62cae3c2963bc8d271] - Code Rule Check OK.

File: os/net/bluetooth/bt_util.h (hunk 1 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_util.h]: removed the space

File: os/net/bluetooth/bt_util.h (hunk 2 above)
Comment (amitpurwar) [reply os/net/bluetooth/bt_util.h]: removed the space

**Comment by amitpurwar** [comment]
...

**Pull request closed without merging**